src/ResourceManager/Sql/Commands.Sql/SqlDatabaseAgent/Cmdlet/Agent/RemoveAzureSqlDatabaseAgent.cs

```diff
@@ -0,0 +1,156 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.SqlDatabaseAgent.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Rest.Azure;
+using System.Globalization;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.SqlDatabaseAgent.Cmdlet
+{
+    /// <summary>
+    /// Defines the Remove-AzureRmSqlDatabaseAgent cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmSqlDatabaseAgent", 
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlDatabaseAgentModel))]
+    public class RemoveAzureSqlDatabaseAgent : AzureSqlDatabaseAgentCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the name of the agent to use.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AgentName")]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent input object model
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlDatabaseAgentModel InputObject { get; set; }
+
+        /// <summary>
+		/// Gets or sets the agent resource id
+		/// </summary>
+		[Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
Same for all cmdlets (won't post this in future ones) - please remove ValueFromPipelineByPropertyName from all properties except InputObject (which is actually ValueFromPipeline) and ResourceId